ticketutil/servicenow.py

```diff
@@ -0,0 +1,321 @@
+"""
+ServiceNow ticket
+
+Provided, you have user and pwd, you can create new ticket like this,
+but you should need other attributes like Category, Item and other non-optional
+attributes from the ServiceNow web GUI.
+
+TODO:
+api PATCH method would enable editing comment
+api DELETE method - maybe, not that necessary
+"""
+
+
+import logging
+import os
+
+import requests
+
+from . import ticket
+
+# TODO ask dranck about credits
+__author__ = 'dranck, rnester, kshirsal, pzubaty'
+
+# Disable warnings for requests because we aren't doing certificate
+# verification
+requests.packages.urllib3.disable_warnings()
+
+DEBUG = os.environ.get('TICKETUTIL_DEBUG', 'False')
+
+if DEBUG == 'True':
+    logging.basicConfig(level=logging.DEBUG)
+else:
+    logging.basicConfig(level=logging.INFO)
+
+
+class ServiceNowTicket(ticket.Ticket):
+    """
+    ServiceNow Ticket object. Contains ServiceNow specific methods for working
+    with tickets.
+    """
+
+    def __init__(self, url, project, auth=None, ticket_id=None):
+        """
+        :param url: ServiceNow service url
+        :param project: ServiceNow table or project
+        :param auth: (<username>, <password>) for HTTP Basic Authentication
+        :param ticked_id: ticked number, eg. PNT1234567
+        """
+        self.ticketing_tool = 'ServiceNow'
+        self.sys_id = None
+
+        # The auth param should be of the form (<username>, <password>) for
+        # HTTP Basic authentication.
+        self.url = url
+        self.auth = auth
+        self.table = project
+        self.rest_url = '{0}/api/now/v1/table/{1}'.format(self.url, self.table)
+        self.auth_url = self.rest_url
+
+        # parent creates request session, ticket_id beforehand causes failure
+        super(ServiceNowTicket, self).__init__(project, ticket_id=None)
+        if ticket_id:
+            super(ServiceNowTicket, self).set_ticket_id(ticket_id)
+
+        # OK for GET, but will it blend? Accept part might be needed later
+        self.s.headers.update({'Content-Type': 'application/json'})
+
+    def process_response(self, response):
+        """
+        Helper function to process Requests response or raise exception
+
+        :param response: response of the Requests query
+        """
+        try:
```

Comment:
This `try/except` block is a bit dangerous. If errors occur you should let them be raised instead of silencing them with logs only.

Also, I think you could vastly simplify this whole method using `response.raise_for_status()` since ServiceNow likely conforms to the status code standards. Perhaps you wouldn't even need the method this way.

Ex:
```python
response.raise_for_status()
return response.json()['result']
```